examples/basic_operations/add_responsive_search_ad.py

```diff
@@ -77,25 +291,240 @@ def main(client, customer_id, ad_group_id):
+            f"{geo_target_constant.resource_name} "
+            f"({geo_target_constant.name}, "
+            f"{geo_target_constant._COUNTRY_CODE}, "
+            f"{geo_target_constant.target_type}, "
+            f"{geo_target_constant.status.name}) "
+            f"is found in _LOCALE ({suggestion._LOCALE}) "
+            f"with reach ({suggestion.reach}) "
+            f"from search term ({suggestion.search_term})."
+        )
+        # Create the campaign criterion for location targeting.
+        campaign_criterion_operation = client.get_type(
+            "CampaignCriterionOperation"
+        )
+        campaign_criterion = campaign_criterion_operation.create
+        campaign_criterion.campaign = campaign_resource_name
+        campaign_criterion.location.geo_target_constant = (
+            geo_target_constant.resource_name
+        )
+        operations.append(campaign_criterion_operation)
+
+    campaign_criterion_service = client.get_service("CampaignCriterionService")
+    campaign_criterion_response = campaign_criterion_service.mutate_campaign_criteria(
+        customer_id=customer_id, operations=[*operations]
+    )
+
+    for result in campaign_criterion_response.results:
+        print(f'Added campaign criterion "{result.resource_name}".')
+
+
+def add_images(client, customer_id, campaign_resource_name):
+    # Step 6.1 - Add Image Asset.
+
+    # Download PNG image from URL.
+    url = "https://gaagl.page.link/bjYi"
+    image_content = requests.get(url).content
+
+    asset_service = client.get_service("AssetService")
+    asset_operation = client.get_type("AssetOperation")
+    asset = asset_operation.create
+    asset.type_ = client.enums.AssetTypeEnum.IMAGE
+
+    # Data field is the raw bytes data of an image.
+    asset.image_asset.data = image_content
+    asset.image_asset.file_size = len(image_content)
+
+    # MIME type of the image (IMAGE_JPEG, IMAGE_PNG, etc.).
+    # See more types on the link below.
+    # https://developers.google.com/google-ads/api/reference/rpc/v11/MimeTypeEnum.MimeType
+    asset.image_asset.mime_type = client.enums.MimeTypeEnum.IMAGE_PNG
+    # Use your favorite image library to determine dimensions
+    asset.image_asset.full_size.height_pixels = 1200
+    asset.image_asset.full_size.width_pixels = 1200
+    asset.image_asset.full_size.url = url
+    # Provide a unique friendly name to identify your asset.
+    # When there is an existing image asset with the same content but a different
+    # name, the new name will be dropped silently.
+    asset.name = f"Testing Image via API {uuid.uuid4()}"
+
+    mutate_asset_response = asset_service.mutate_assets(
+        customer_id=customer_id, operations=[asset_operation]
+    )
+    print("Uploaded file(s):")
+    for row in mutate_asset_response.results:
+        print(f"\tResource name: {row.resource_name}")
+
+    image_asset_resource_name = mutate_asset_response.results[0].resource_name
+
+    # Step 6.2 - Create Image Extension
```

Comment:
Why are we including this example of extensions? As of 3 April 2023 we want people to stop doing this. Some users, if they are early in the migration process will no longer be able to do this.

I recommend not including any code that deals with extensions.

Reply:
Recall we discussed offline that these changes are just a renaming existing example that we recently approved (https://github.com/googleads/google-ads-python/pull/715).

We'll update this example (along with others) so that it uses assets around when the April 3 migration happens, for now this does illustrate the correct workflow for this type of functionality, right?